File: js/packages/problems/src/lib/p_217_contains_duplicate.ts
export function containsDuplicate(nums: number[]): boolean {
    /**
     * Contains the individuel numbers that were found in the 'nums' array
     */
    const cache_memory: number[] = []

    /**
     * The length of the 'nums' array
     */
    const arrayLength = nums.length

    // START Iterate over the 'nums' array
    for (let i = 0; i < arrayLength; i++) {
        if (cache_memory.includes(nums[i])) {
            return true
        }
        else {
            cache_memory.push(nums[i])
        }
    }
    // END Iterate over the 'nums' array

    return false;
}

File: py/problems/p_217_contains_duplicate.py
"""
Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.
"""
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        """
        Checks if a list contains a duplicate
        """

        # contains the numbers that are saved
        memory_cache = []
        
        for item in nums:
            # If we find the number in the cache
            if item in memory_cache:
                return True
            
            # Save the new number in the cache
            memory_cache.append(item)

        return False